resolver/resolver.cc

```diff
@@ -2649,6 +2649,36 @@ class ResolveSignaturesWalk {
                 j++;
             }
         }
+
+        // Later passes are going to separate the sig and the method definition.
+        // Record some information in the sig call itself so that we can reassociate
+        // them later.
+        auto *send = sig.origSend;
+        auto &origArgs = send->args;
+        auto *self = ast::cast_tree<ast::Local>(send->args[0]);
+        if (self == nullptr) {
+            return;
+        }
+
+        // We distinguish "user-written" sends by checking for self.
+        // T::Sig::WithoutRuntime.sig wouldn't have any runtime effect that we need
+        // to record later.
+        if (self->localVariable != core::LocalVariable::selfVariable()) {
+            return;
+        }
+
+        auto *cnst = ast::cast_tree<ast::ConstantLit>(send->recv);
+        ENFORCE(cnst != nullptr);
+        // This condition really shouldn't happen.
+        if (cnst == nullptr) {
+            return;
+        }
+
+        cnst->symbol = core::Symbols::Sorbet_Private_Static_ResolvedSig();
+
+        origArgs.emplace_back(mdef.flags.isSelfMethod ? ast::MK::True(send->loc) : ast::MK::False(send->loc));
+        origArgs.emplace_back(ast::MK::Symbol(send->loc, method.data(ctx)->name));
+        send->numPosArgs += 2;
```

Comment:
This is only possible here because of this recent change:

https://github.com/sorbet/sorbet/pull/4294

my understanding was that PR was a performance-only change.

Combined with that fact that we're using raw pointers to cross-talk across multiple phases, I'd love to either find a way to do this that involves copying the things we need (so that we can do it without mutating), or at the very least write a comment on top of the `origSend` pointer outlining our assumptions about this phase (phase that parses sigs doesn't mutate them, they're never reallocated so the pointer is valid until this phases, this phase is already serial because it mutates global state, etc.)